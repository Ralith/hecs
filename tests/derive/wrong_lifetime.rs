use hecs::Query;

#[derive(Query)]
struct Foo<'a> {
    foo: &'a i32,
    bar: &'static mut bool,
}

fn main() {}
