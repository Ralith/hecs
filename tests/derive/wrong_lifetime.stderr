error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
 --> $DIR/wrong_lifetime.rs:3:10
  |
3 | #[derive(Query)]
  |          ^^^^^
  |
note: first, the lifetime cannot outlive the lifetime `'a` as defined here...
 --> $DIR/wrong_lifetime.rs:3:10
  |
3 | #[derive(Query)]
  |          ^^^^^
note: ...so that the types are compatible
 --> $DIR/wrong_lifetime.rs:3:10
  |
3 | #[derive(Query)]
  |          ^^^^^
  = note: expected `Fetch<'a>`
             found `Fetch<'_>`
  = note: but, the lifetime must be valid for the static lifetime...
note: ...so that reference does not outlive borrowed content
 --> $DIR/wrong_lifetime.rs:3:10
  |
3 | #[derive(Query)]
  |          ^^^^^
  = note: this error originates in the derive macro `Query` (in Nightly builds, run with -Z macro-backtrace for more info)
